pkg/cluster/cluster.go

```diff
@@ -295,6 +295,8 @@ func (c *Cluster) run(stopC <-chan struct{}) {
 				break
 			}
 
+			garbagecollection.CollectDeadMemberServices(c.cluster.Name, c.cluster.Namespace, c.config.KubeCli, c.members, c.logger)
```

Comment:
> gc does not need to change membership. it only reads from membership struct.

We can't have another routine changing the membership while gc reading it. For example, gc might read members [A, B, C], but it's changed to [B, C, D]. Then it will delete D, but D is valid for that moment.